addons/mail/static/tests/helpers/mock_server.js

```diff
@@ -2176,4 +2181,32 @@ MockServer.include({
             starred_counter: this._getRecords('mail.message', [['starred_partner_ids', 'in', user.partner_id]]).length,
         };
     },
+    /**
+     * Simulate the `notify_cancel_by_type` on `mail.thread` .
+     * Note that this method is overridden by snailmail module but not simulated here.
+     */
+    _mockMailThreadNotifyCancelByType(model, notificationType) {
+        const self = this;
+        const authorId = this.currentPartnerId;
+
+        let notificationMails = this._getRecords('mail.notification', [
+            ['notification_type', '=', notificationType],
+            ['notification_status', 'in', ['bounce', 'exception']],
+        ]).map(notification => ({
+            notification: notification,
+            mail: self._getRecords('mail.message', [['id', '=', notification.mail_message_id]])[0]
+        })).filter(notifMail => notifMail.mail.model === model && notifMail.mail.author_id === authorId);
```

Comment:
Have the filter inside the get records domain if possible? (should be possible in python, but the mock server domain might not be good enough)

Reply:
I think it is not possible but I may be wrong. Here I need to get the mail messages related to the selected notifications and do additional filtering on values of those mail message.